File: src/main/java/github/cweijan/ultimate/generator/GeneratorAdapter.java
package github.cweijan.ultimate.generator;

import github.cweijan.ultimate.generator.impl.MysqlGenerator;
import github.cweijan.ultimate.db.config.DbConfig;

import java.util.Objects;

/**
 根据DB配置获取sql生成器
 */
public class GeneratorAdapter{

    private String driverName;

    public GeneratorAdapter(DbConfig config){

        this.driverName =config.getDriver();
    }

    public SqlGenerator getGenerator(){

        Objects.requireNonNull(driverName);
        if("com.mysql.jdbc.Driver".equals(driverName)){
            return new MysqlGenerator();
        }

        return null;
    }
}

File: src/main/java/github/cweijan/ultimate/generator/GeneratorAdapter.kt
package github.cweijan.ultimate.generator

import github.cweijan.ultimate.generator.impl.MysqlGenerator
import github.cweijan.ultimate.generator.impl.OracleGenerator
import github.cweijan.ultimate.generator.impl.PostgresqlGenerator

/**
 * 根据DB配置获取sql生成器
 */
object GeneratorAdapter {

    fun getSqlGenerator(driverName: String?): SqlGenerator {
        return when (driverName) {
            DriverConstant.MYSQL_DRIVER_NAME -> MysqlGenerator()
            DriverConstant.ORACLE_DRIVER_NAME -> OracleGenerator()
            DriverConstant.POSTGRESQL_DRIVER_NAME -> PostgresqlGenerator()
            DriverConstant.SQLITE_DRIVER_NAME -> PostgresqlGenerator()
            else -> MysqlGenerator()
        }
    }

}
